enterprise/j2ee.sun.dd/test/unit/src/org/netbeans/modules/j2ee/sun/dd/api/DDProviderTest.java

```diff
@@ -42,108 +43,71 @@ public DDProviderTest() {
         super("DDProvider");
     }
 
-    @Before
-    public void setUp() {
-    }
-
-    @After
-    public void tearDown() {
-    }
-
     @Test
     public void testGetEjbDDRoot() throws FileNotFoundException, IOException, SAXException {
         File[] list = getDataDir().listFiles();
         if (null == list || list.length == 0)
             return;
+        SunEjbJar root = null;
         for (File fo : list) {
             if (fo.getName().startsWith("valid-sun-ejb-jar")) {
-                InputStream is = null;
-                try {
-                    is = new FileInputStream(fo);
-
-                    DDProvider.getDefault().getEjbDDRoot(new InputSource(is));
-                } finally {
-                    if (null != is) {
-                        try {
-                            is.close();
-                        } catch (IOException ioe) { }
-                    }
+                try (InputStream is = new FileInputStream(fo)) {
+                    root = DDProvider.getDefault().getEjbDDRoot(new InputSource(is));
+                    assertNotNull(root);
                 }
             }
         }
-
-
+        assertNotNull(root);
     }
 
     @Test
     public void testGetWebDDRoot() throws FileNotFoundException, IOException, SAXException, DDException {
         File[] list = getDataDir().listFiles();
         if (null == list || list.length == 0)
             return;
+        SunWebApp root = null;
         for (File fo : list) {
             if (fo.getName().startsWith("valid-sun-web")) {
-                InputStream is = null;
-                try {
-                    is = new FileInputStream(fo);
-
-                    DDProvider.getDefault().getWebDDRoot(new InputSource(is));
-                } finally {
-                    if (null != is) {
-                        try {
-                            is.close();
-                        } catch (IOException ioe) { }
-                    }
+                try (InputStream is = new FileInputStream(fo)) {
+                    root = DDProvider.getDefault().getWebDDRoot(new InputSource(is));
+                    assertNotNull(root);
                 }
             }
         }
-
+        assertNotNull(root);
     }
 
     @Test
-    public void testGetAppDDRoot() throws FileNotFoundException, IOException, SAXException {
+    public void fails_testGetAppDDRoot() throws FileNotFoundException, IOException, SAXException {
         File[] list = getDataDir().listFiles();
         if (null == list || list.length == 0)
             return;
+        SunApplication root = null;
         for (File fo : list) {
             if (fo.getName().startsWith("valid-sun-application")) {
-                InputStream is = null;
```

Comment:
fyi: those two test cases did pass even though the `if-block` never matched anything. Essentially a false positive.

fixed the tests and turned them off